pandas/sparse/frame.py

```diff
@@ -187,9 +213,47 @@ def _init_matrix(self, data, index, columns, dtype=None):
         if len(index) != N:
             raise ValueError('Index length mismatch: %d vs. %d' %
                              (len(index), N))
+        return index, columns
 
-        data = dict([(idx, data[:, i]) for i, idx in enumerate(columns)])
-        return self._init_dict(data, index, columns, dtype)
+    def to_coo(self):
+        """
+        Convert the frame to its SciPy sparse COO matrix representation.
```

Comment:
Oh, at first glance, it's much simpler!

In your case:
```py
>>> ss.to_coo()
---------------------------------------------------------------------------
ValueError: Is not a partition because union is not the whole.

>>> # huh ???
```
With proposed (shown only for 2d case):
```py
>>> from scipy.sparse import coo_matrix
>>> sdf = pd.SparseDataFrame(coo_matrix([[0, 1, 0],
...                                      [1, 0, 0]]))

>>> sdf  # shit is a dataframe
     0    1   2
0  NaN  1.0 NaN
1  1.0  NaN NaN

>>> sdf[0]  # behaves like a dataframe
0    NaN
1    1.0
Name: 0, dtype: float64
BlockIndex
Block locations: array([1], dtype=int32)
Block lengths: array([1], dtype=int32)

>>> sdf.to_coo()  # no complex arguments
<2x3 sparse matrix of type '<class 'numpy.float64'>'
	with 2 stored elements in COOrdinate format>

>>> # gives back the kind of 2d matrix it was given
>>> # no COOrdinate (multi)index computing by hand
>>> sdf[[0, 1]].to_coo().toarray()  # just works
array([[0, 1],
       [1, 0]])
```
That was the idea, at least. :smiley: